### PR TITLE
Admin room name change and added topic

### LINKED ISSUE #198: IRC Admin Room -> Network-specific Name
Ideas:
- Need to have the first two words to distinguish between admin rooms.
- Probably shouldn't have the word "room" in the name, it's already a room.
- Maybe a topic should describe what it does!

Candidates:
- ~~Freenode Admin Room~~ includes word room
- ~~Freenode IRC Admin Room~~ includes word room
- ~~IRC Admin (Freenode)~~ can't distinguish with first two words
- ~~irc.freenode.net Connection Admin~~ too long, use topic instead
- **irc.freenode.net Admin, /w topic**

### PR DESCRIPTION
**Pull request by lukebarnard1**
Example:
 - Room name: 'irc.freenode.net Admin'
 - Room topic: 'This room shows any errors or status messages from irc.freenode.net, as well as letting you control the connection.'

Fixes #198

### CODE REVIEW

File: lib/bridge/IrcHandler.js [hunk 1]
```diff
@@ -443,7 +443,10 @@ IrcHandler.prototype.onMetadata = Promise.coroutine(function*(req, client, msg)
         let response = yield this.ircBridge.getAppServiceBridge().getIntent().createRoom({
             createAsClient: false,
             options: {
-                name: "IRC Admin Room",
+                name: `${client.server.domain} Admin`,
```
Comment (Kegsay) [inline lib/bridge/IrcHandler.js]: I like this. The only problem I see is that the domains may end up being ugly (think `ipv6.chat.freenode.net` or equiv). Should we instead prioritise the "description" field in the config or whatever it was? That would then be "Freenode".

Comment (Kegsay) [inline lib/bridge/IrcHandler.js]: Though "Freenode Admin" is also pretty confusing. Ugh, naming is hard.

File: lib/bridge/IrcHandler.js [hunk 2]
```diff
@@ -443,7 +443,10 @@ IrcHandler.prototype.onMetadata = Promise.coroutine(function*(req, client, msg)
...
+                topic:  "This room shows any errors or status messages from " +
+                        "${client.server.domain}, as well as letting you control " +
```
Comment (lukebarnard1) [inline lib/bridge/IrcHandler.js]: oh.. I thought I had, I'll test again

File: lib/bridge/IrcHandler.js (hunk 1 above)
Comment (lukebarnard1) [inline lib/bridge/IrcHandler.js]: "Admin" sounds like you are the administrator for the network... Or at least in a room that connects you to an admin user.

I had other suggestions in #198

Comment (Kegsay) [inline lib/bridge/IrcHandler.js]: How about keeping it more informal perhaps, and something like "Your Freenode Connection"?

Comment (Kegsay) [inline lib/bridge/IrcHandler.js]: Really need @ara4n to chime in on this.

**Comment by Kegsay** [comment]
LGTM aside from naming fun times!

File: lib/irc/IrcServer.js [hunk 3]
```diff
@@ -17,6 +17,25 @@ function IrcServer(domain, serverConfig) {
     this.config = serverConfig;
 }
 
+/**
+ * Get a string that represents the human-readable name for a server.
+ * @returns config.name if available, otherwise it will attempt to
```
Comment (Kegsay) [inline lib/irc/IrcServer.js]: If you're going to use JSDoc then please make it valid JSDoc:
```js
/**
 * @return {string} config.name blah blah
 */
```

**Comment by Kegsay** [comment]
Can we deprecate `description` and return `name` in the 3P lookup responses as well then please?

**Comment by Kegsay** [comment]
Code-wise, LGTM

**Comment by lukebarnard1** [comment]
Done. Unfortunately, this will mean changing the descriptions of all the bridges unless you want admin rooms with names like `Freenode (irc.freenode.net) IRC Bridge status` /o\

**Pull request merged**